flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -536,6 +553,9 @@ private void openNewPartFile(Path bucketPath, BucketState<T> bucketState) throws
 			partPath = new Path(bucketPath, partPrefix + "-" + subtaskIndex + "-" + bucketState.partCounter);
 		}
 
+		// Record the creation time of the bucket
+		bucketState.firstWrittenToTime = processingTimeService.getCurrentProcessingTime();
```

Comment:
rename parameter to `bucket.creationTime`? At this point, nothing has been written to the file. Actually, the file has not even been created yet.

Reply:
I think the current behavior is also more consistent than starting the timeout when the first record is written.